src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -106,7 +107,10 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
         val files = location.allFiles
         // Note that source files are currently fingerprinted when the optimized plan is
         // fingerprinted by LogicalPlanFingerprint.
-        val sourceDataProperties = Hdfs.Properties(Content.fromLeafFiles(files).get)
+        val sourceDataProperties =
+          Hdfs.Properties(Content.fromLeafFiles(files, fileIdTracker).get)
+        assert(fileIdTracker.getMaxFileId != IndexConstants.INITIAL_FILE_ID)
```

Comment:
It asserts that as we list source files, the id is actually generated for them (FileIdTracker should not be empty after going over all source files for the index).

Reply:
Shouldn't this be just checked with unit tests? BTW, this assert also has a dependency on `files` being non empty as well.

Reply:
@imback82 This is already checked with unit tests and yes there is dependency on `files`. Adding this assert is really a double check here, before proceeding. It can be dropped if you believe it is simply redundant/expensive.

Reply:
Cool, in that case, can we just drop it? Thanks!